src/windows.rs

```diff
@@ -0,0 +1,307 @@
+                ptr::null_mut());
+
+            if handle == INVALID_HANDLE_VALUE {
+                let err = if watching_file {
+                    Err(Error::Generic("You attempted to watch a single file, but parent directory could not be opened.".to_owned()))
+                } else {
+                    // TODO: Call GetLastError for better error info?
+                    Err(Error::PathNotFound)
+                };
+                let _ = self.tx.send(Event {
+                    path: None,
+                    op: err
+                });
+                return;
+            }
+        }
+        let wf = if watching_file {
+            Some(path.clone())
+        } else {
+            None
+        };
+        let rd = ReadData {
+            dir: dir_target,
+            file: wf
+        };
+        self.watches.insert(path.clone(), handle);
+        start_read(&rd, &self.tx, handle);
+    }
+
+    fn remove_watch(&mut self, path: PathBuf) {
+        if let Some(handle) = self.watches.remove(&path) {
+            unsafe {
+                close_handle(handle);
+            }
+        }
+    }
+}
+
+unsafe fn close_handle(handle: HANDLE) {
+    // TODO: Handle errors
+    kernel32::CancelIo(handle);
+    kernel32::CloseHandle(handle);
+}
+
+fn start_read(rd: &ReadData, tx: &Sender<Event>, handle: HANDLE) {
+    let mut request = Box::new(ReadDirectoryRequest {
+        tx: tx.clone(),
+        handle: handle,
+        buffer: [0u8; BUF_SIZE as usize],
+        data: rd.clone()
+    });
+
+    let flags = winnt::FILE_NOTIFY_CHANGE_FILE_NAME
+              | winnt::FILE_NOTIFY_CHANGE_DIR_NAME
+              | winnt::FILE_NOTIFY_CHANGE_ATTRIBUTES
+              | winnt::FILE_NOTIFY_CHANGE_SIZE
+              | winnt::FILE_NOTIFY_CHANGE_LAST_WRITE
+              | winnt::FILE_NOTIFY_CHANGE_CREATION
+              | winnt::FILE_NOTIFY_CHANGE_SECURITY;
+
+    let monitor_subdir = if (&request.data.file).is_none() {
+      1
+    } else {
+      0
+    };
+
+    unsafe {
+        let mut overlapped: Box<OVERLAPPED> = Box::new(mem::zeroed());
+        // When using callback based async requests, we are allowed to use the hEvent member
+        // for our own purposes
+
+        let req_buf = request.buffer.as_mut_ptr() as *mut c_void;
+        let request_p: *mut c_void = mem::transmute(request);
+        overlapped.hEvent = request_p;
+
+        // This is using an asynchronous call with a completion routine for receiving notifications
+        // An I/O completion port would probably be more performant
+        kernel32::ReadDirectoryChangesW(
+            handle,
+            req_buf,
+            BUF_SIZE,
+            monitor_subdir,
+            flags,
+            &mut 0u32 as *mut u32,  // This parameter is not used for async requests
+            &mut *overlapped as *mut OVERLAPPED,
+            Some(handle_event));
+
+        mem::forget(overlapped);
```

Comment:
Could do that; but in my other branch I check the return value here, because It can fail.  In the fail case you have to re-box the memory to get a drop:
https://github.com/jmquigs/rsnotify/blob/7dc1f555ff065154eb9e57134631e52bb958c6dc/src/windows.rs#L231
...maybe I should just include those changes here